### PR TITLE
[10466] Fix for Missing headers in request body

### PR DESCRIPTION
**Pull request by aman-v-singh**
## Description

- [10466](https://github.com/postmanlabs/postman-app-support/issues/10466) Headers were missing while importing WSDL SOAP collections.

## Sample Input and Output

- Input: [AppWSDL.txt](https://github.com/user-attachments/files/17644532/AppWSDL.txt)
- Output -  Headers are present
Screenshot 2024-11-21 at 6 25 46 PM

### Checklist:

- [x] Checked all the config values
- [x] Added relevant unit tests for my code

### CODE REVIEW

File: lib/WsdlToPostmanCollectionMapper.js [hunk 1]
```diff
@@ -339,7 +339,7 @@ class WsdlToPostmanCollectionMapper {
   createItemsFromOperations(operations, urlVariablesData, securityPolicyArray, xmlParser) {
     let items = operations.map((operation) => {
       let requestBody = this.wsdlToPostmanCollectionBodyMapper
-          .getBody(operation, operation.input[0], securityPolicyArray, xmlParser),
+          .getBody(operation, operation.input[0], [operation.header, securityPolicyArray], xmlParser),
```
Comment (VShingala) [inline lib/WsdlToPostmanCollectionMapper.js:342]: why not add separate arg for `operation.header` instead using array?

**Review by VShingala — CHANGES_REQUESTED** [review]
Let's add few more tests, ideally separate unit test for this instead using existing test to describe test case better

File: lib/WsdlToPostmanCollectionMapper.js (hunk 1 above)
Comment (aman-v-singh) [reply lib/WsdlToPostmanCollectionMapper.js:342]: I would have had to change every function in between `wsdlPostmanCollectionMapper.getBody` to `SOAPMessageHelper.convertInputToMessage`, so I decided to make it into an array as it is only processed at the end at `SOAPMessageHelper.convertInputToMessage`.

File: test/data/validWSDLs11/elementFormDefaultQualified.wsdl [hunk 2]
```diff
@@ -285,6 +285,11 @@
                     <xsd:element name="Xpath" type="xsd:string" minOccurs="0" maxOccurs="1" />
                 </xsd:sequence>
             </xsd:complexType>
+            <xsd:complexType name="Workday_Common_HeaderType">
+                <xsd:sequence>
+                <xsd:element name="Include_Reference_Descriptors_In_Response" type="xsd:boolean" minOccurs="0" maxOccurs="1"/>
```
[the remark this answers is not in the document]
Comment (aman-v-singh) [reply test/data/validWSDLs11/elementFormDefaultQualified.wsdl:290]: @VShingala I have added relevant tests.

File: lib/WsdlToPostmanCollectionMapper.js (hunk 1 above)
Comment (VShingala) [reply lib/WsdlToPostmanCollectionMapper.js:342]: @aman-v-singh This is not advisable practice. i.e. We should not be changing the argument type whenever required and especially not to array. Let's either use a separate argument for this or use Object as org where you can pass multiple props for better future extensibility.

e.g. something like
```
getBody(operation, operation.input[0], { headers: operation.header, securityPolicyArray }, xmlParser)
```

File: test/data/validWSDLs11/soapBodyAuthHeaders.wsdl [hunk 3]
```diff
@@ -0,0 +1,83 @@
+<wsdl:definitions xmlns:s="http://www.w3.org/2001/XMLSchema" xmlns:soap12="http://schemas.xmlsoap.org/wsdl/soap12/" xmlns:http="http://schemas.xmlsoap.org/wsdl/http/" xmlns:mime="http://schemas.xmlsoap.org/wsdl/mime/" xmlns:tns="http://localhost/App.asmx" xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/" xmlns:tm="http://microsoft.com/wsdl/mime/textMatching/" xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:wsse="http://schemas.xmlsoap.org/ws/2004/09/policy" targetNamespace="http://localhost/App.asmx">
```
Comment (VShingala) [inline test/data/validWSDLs11/soapBodyAuthHeaders.wsdl:1]: Let's keep this file properly indented for better readability.

**Review by VShingala — APPROVED** [review]

**Pull request merged**